test/test-genesis.ts

```diff
@@ -384,4 +385,66 @@ describe('Genesis Contract', () => {
       contract.connect(owner).mintReservedGods(whitelistedAddr, 10),
     ).to.be.revertedWith('Not enough reserved gods left');
   });
+
+  it('Contract base URI is unrevealed URI if not changed', async function () {
+    await contract.connect(owner).unpause();
+    expect(await contract.tokenURI(1)).to.be.equal(constants.unrevealedURI);
+  });
+
+  it('Contract base URI is unrevealed URI if not changed', async function () {
+    await contract.connect(owner).unpause();
+    expect(await contract.tokenURI(1)).to.be.equal(constants.unrevealedURI);
+    expect(await contract.tokenURI(100)).to.be.equal(constants.unrevealedURI);
+    expect(await contract.tokenURI(4000)).to.be.equal(constants.unrevealedURI);
+  });
+
+  it('Contract base URI is revealed URI if set', async function () {
+    await contract.connect(owner).unpause();
+    await contract.connect(owner).setBaseURI(constants.revealedURI);
+
+    expect(await contract.tokenURI(1)).to.be.equal(
+      `${constants.unrevealedURI}1`,
+    );
+    expect(await contract.tokenURI(100)).to.be.equal(
+      `${constants.unrevealedURI}100`,
+    );
+
+    expect(await contract.tokenURI(0)).to.be.equal(
+      `${constants.unrevealedURI}0`,
+    );
+    expect(await contract.tokenURI(9999)).to.be.equal(
+      `${constants.unrevealedURI}9999`,
+    );
+  });
+
+  it('Make sure that the metadata is not available until before the collection is revealed', async function () {
+    await contract.connect(owner).unpause();
+    await expect(
+      contract.connect(owner).getMetadataForTokenId(1),
+    ).to.be.revertedWith('Not revealed yet');
+  });
+
+  it('No metadata is returned if trying to access an invalid token id', async function () {
+    await contract.connect(owner).unpause();
+    await contract.connect(owner).setBaseURI(constants.revealedURI);
+    await expect(
+      contract.connect(owner).getMetadataForTokenId(0),
+    ).to.be.revertedWith('Invalid tokenId');
+    await expect(
+      contract.connect(owner).getMetadataForTokenId(1001),
+    ).to.be.revertedWith('Invalid tokenId');
+    await expect(
+      contract.connect(owner).getMetadataForTokenId(4000),
+    ).to.be.revertedWith('Invalid tokenId');
+  });
+
+  it('Metadata is returned if trying to access a valid token id', async function () {
```

Comment:
We could add if trying to access a valid token id and URI is revealed.

Reply:
it's exactly this test?

Reply:
i meant in the description. 
`Metadata is returned if trying to access a valid token id and collection is revealed`

Reply:
got it. updated